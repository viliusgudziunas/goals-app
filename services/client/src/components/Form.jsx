import React, { useState } from 'react';
import PropTypes from 'prop-types';

const Form = ({ formType }) => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const handleFormSubmit = e => {
    e.preventDefault();
    console.log(email);
    console.log(password);
  };

  return (
    <div className='container'>
      <br />
      {formType === 'Login' && <h1>Log In</h1>}
      {formType === 'Register' && <h1>Register</h1>}
      <hr />
      <div className='row'>
        <form className='col-md-4' onSubmit={handleFormSubmit}>
          <div className='form-group'>
            <input
              name='email'
              type='email'
              className='form-control'
              placeholder='Enter an email address'
              id='formEmail'
              value={email}
              onChange={e => setEmail(e.target.value)}
              required
            />
          </div>
          <div className='form-group'>
            <input
              name='passowrd'
              type='password'
              className='form-control'
              placeholder='Enter a password'
              id='formPassword'
              value={password}
              onChange={e => setPassword(e.target.value)}
              required
            />
          </div>
          <button type='submit' className='btn btn-success btn-block'>
            Submit
          </button>
        </form>
      </div>
    </div>
  );
};

Form.propTypes = {
  formType: PropTypes.string.isRequired
};

export default Form;
