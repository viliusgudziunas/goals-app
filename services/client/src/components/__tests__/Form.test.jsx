import React from 'react';
import { shallow } from 'enzyme';

import Form from '../Form';

const forms = [
  { type: 'Register', title: 'Register' },
  { type: 'Login', title: 'Log In' }
];

forms.forEach(formInfo => {
  describe(`${formInfo.title} <Form />`, () => {
    const wrapper = shallow(<Form formType={formInfo.type} />);
    it('should match the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('should have a header', () => {
      expect(wrapper.find('h1').length).toBe(1);
      expect(wrapper.find('h1').get(0).props.children).toBe(formInfo.title);
    });

    const form = wrapper.find('form');
    it('should have a form', () => {
      expect(form.length).toBe(1);
    });

    const formFields = form.find('input');
    it('should have two input fields', () => {
      expect(formFields.length).toBe(2);
    });

    it('should have email field', () => {
      expect(formFields.get(0).props.type).toBe('email');
      expect(formFields.get(0).props.value).toBe('');
    });

    it('should have password field', () => {
      expect(formFields.get(1).props.type).toBe('password');
      expect(formFields.get(1).props.value).toBe('');
    });

    it('should have a submit button', () => {
      expect(form.find('button').length).toBe(1);
      expect(form.find('button').get(0).props.type).toBe('submit');
    });
  });
});
