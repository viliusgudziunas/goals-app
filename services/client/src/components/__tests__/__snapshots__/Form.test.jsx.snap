// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Log In <Form /> should match the snapshot 1`] = `"<div class=\\"container\\"><br/><h1>Log In</h1><hr/><div class=\\"row\\"><form class=\\"col-md-4\\"><div class=\\"form-group\\"><input type=\\"email\\" name=\\"email\\" class=\\"form-control\\" placeholder=\\"Enter an email address\\" id=\\"formEmail\\" value=\\"\\" required=\\"\\" minLength=\\"6\\"/></div><div class=\\"form-group\\"><input type=\\"password\\" name=\\"passowrd\\" class=\\"form-control\\" placeholder=\\"Enter a password\\" id=\\"formPassword\\" value=\\"\\" required=\\"\\" minLength=\\"4\\"/></div><button type=\\"submit\\" class=\\"btn btn-success btn-block\\">Submit</button></form></div></div>"`;

exports[`Register <Form /> should match the snapshot 1`] = `"<div class=\\"container\\"><br/><h1>Register</h1><hr/><div class=\\"row\\"><form class=\\"col-md-4\\"><div class=\\"form-group\\"><input type=\\"email\\" name=\\"email\\" class=\\"form-control\\" placeholder=\\"Enter an email address\\" id=\\"formEmail\\" value=\\"\\" required=\\"\\" minLength=\\"6\\"/></div><div class=\\"form-group\\"><input type=\\"password\\" name=\\"passowrd\\" class=\\"form-control\\" placeholder=\\"Enter a password\\" id=\\"formPassword\\" value=\\"\\" required=\\"\\" minLength=\\"4\\"/></div><button type=\\"submit\\" class=\\"btn btn-success btn-block\\">Submit</button></form></div></div>"`;
